### PR TITLE
Refactor grades activity

### PR DESCRIPTION
**Pull request by thellmund**
We had crashes in GradesActivity due to nullability. I converted the class to Kotlin and improved the code.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+    private var exams: List<Exam>? = null
+    private var programIds: List<String>? = null
+
+    private var spinnerPosition = 0
+
+    private var isFetched = false
+
+    // everything for the charts
+    private var barMenuItem: MenuItem? = null
+    private var pieMenuItem: MenuItem? = null
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: lateinit?

I don't think we have a use-case where those being null makes sense?

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+import java.text.NumberFormat
+import java.util.*
+
+/**
+ * Activity to show the user's grades/exams passed.
+ */
+class GradesActivity : ActivityForAccessingTumOnline<ExamList>(R.layout.activity_grades) {
+
+    // exams data and list
+    private var exams: List<Exam>? = null
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: `exams` is passed around as argument several times and assigned but never read as member variable.

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+    private var programIds: List<String>? = null
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: same here

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        }
+        return gradeDistribution
+    }
+
+    /**
+     * Initialize the spinner for choosing between the study programs.
+     */
+    private fun initSpinner(exams: List<Exam>) {
+        // Set Spinner data
+        val filters = arrayListOf<String>(getString(R.string.all_programs))
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: I'd use `mutableListOf()`

Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: Yep, that better describes the intentions 👍

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt (hunk 1 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: We’ve had NullPointerExceptions due to them being null

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+        instanceState.putInt(KEY_SPINNER_POSITION, spinnerPosition)
+    }
+
+    /**
+     * Toggles between list view and chart view in landscape mode.
+     */
+    fun toggleInLandscape(view: View) {
+        val showChart = chartsContainer.visibility == View.GONE
+
+        //swipeRefreshLayout.visibility = if (showChart) View.GONE else View.VISIBLE
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt]: :bell::bell::bell: That's not a real comment

**Pull request merged**